packages/catalogue/src/catalogue.ts

```diff
@@ -0,0 +1,29 @@
+import type { CredentialsProvider } from '@tidal-music/common';
+import createClient from 'openapi-fetch';
+
+import type { paths } from './catalogueAPI';
+
+let authToken: string | undefined = undefined;
+
+/**
+ * Initializes the client with the provided credentials.
+ *
+ * @param credentialsProvider The credentials provider.
+ */
+export async function initCatalogueClient(
+  credentialsProvider: CredentialsProvider,
+) {
+  const credentials = await credentialsProvider.getCredentials();
+  authToken = credentials.token;
+  //TODO: listen for changes to the token
+}
+
+export const catalogueClient = createClient<paths>({
+  baseUrl: 'https://openapi.tidal.com/',
+  headers: {
+    get Authorization() {
+      return authToken ? `Bearer ${authToken}` : undefined;
+    },
```

Comment:
Interesting, I would have put the `.getCredentials` in here, but I assume it can't be async?

Reply:
The lib was just updated with support for middlewares: https://openapi-ts.pages.dev/openapi-fetch/middleware-auth so will be a bit cleaner when I update it to that :-)

Reply:
Switched to the Middleware approach, so now it can be async :-)

Reply:
Looks great with the middleware! 🎉